src/test/pmemlog_open/pmemlog_open.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmemlog_open.c -- unit test for opening a log memory pool
+ *
+ * usage: pmemlog_open path
+ */
+
+#include "unittest.h"
+
+#define	SIZEOF_TESTFILE	(64 * 1024 * 1024)
+#define	CREATE_MODE		(0664)
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "pmemlog_open");
+
+	if (argc != 2)
+		FATAL("usage: %s path", argv[0]);
+
+	PMEMlogpool *handle;
+	const char *path = argv[1];
+
+	if (strcmp(path, "NULLFILE") == 0) {
+		if ((handle = pmemlog_create("./testfile",
+				SIZEOF_TESTFILE, CREATE_MODE)) == NULL) {
+			OUT("!./testfile: pmemlog_create");
+			goto err;
+		}
+
+		if ((handle = pmemlog_open("./testfile")) == NULL) {
+			OUT("!./testfile: pmemlog_open");
+			goto err;
+		}
+	} else {
+		if ((handle = pmemlog_create(path,
+				0, CREATE_MODE)) == NULL) {
+			OUT("!%s: pmemlog_create", path);
+			goto err;
+		}
+
+		if ((handle = pmemlog_open(path)) == NULL) {
+			OUT("!%s: pmemlog_open", path);
+			goto err;
+		}
+	}
```

Comment:
I don't think I'm getting why the pmemlog_open() test always calls pmemlog_create().  And you're opening the pool twice, trashing the handle returned from create with the one returned by open.

Might be time to step back and decide what the cases are that you're testing, maybe list them in the README file ro something, and then come up with the simple program that allows those cases to be tested.  Or maybe you've done that and I'm just not connecting the dots.